*** map.sh.org
--- map.sh
***************
*** 26,36 ****
  	local link="map-$cfg"
  
  	# uncomment for legacy MAP0 mode
! 	#export LEGACY=1
  
! 	local type mtu ttl tunlink zone encaplimit
  	local rule ipaddr ip4prefixlen ip6prefix ip6prefixlen peeraddr ealen psidlen psid offset
! 	json_get_vars type mtu ttl tunlink zone encaplimit
  	json_get_vars rule ipaddr ip4prefixlen ip6prefix ip6prefixlen peeraddr ealen psidlen psid offset
  
  	[ "$zone" = "-" ] && zone=""
--- 26,36 ----
  	local link="map-$cfg"
  
  	# uncomment for legacy MAP0 mode
! 	export LEGACY=1
  
! 	local type mtu ttl tunlink zone encaplimit snatstartps snatendps
  	local rule ipaddr ip4prefixlen ip6prefix ip6prefixlen peeraddr ealen psidlen psid offset
! 	json_get_vars type mtu ttl tunlink zone encaplimit snatstartps snatendps
  	json_get_vars rule ipaddr ip4prefixlen ip6prefix ip6prefixlen peeraddr ealen psidlen psid offset
  
  	[ "$zone" = "-" ] && zone=""
***************
*** 140,146 ****
--- 140,149 ----
  	      json_add_string snat_ip $(eval "echo \$RULE_${k}_IPV4ADDR")
  	    json_close_object
  	  else
+ 	    local pscount=0
  	    for portset in $(eval "echo \$RULE_${k}_PORTSETS"); do
+ 	     pscount=`expr $pscount + 1`
+ 	     if [ "$pscount" -ge "${snatstartps:-1}" -a "$pscount" -le "${snatendps:-4096}" ]; then
                for proto in icmp tcp udp; do
  	        json_add_object ""
  	          json_add_string type nat
***************
*** 152,157 ****
--- 155,161 ----
                    json_add_string snat_port "$portset"
  	        json_close_object
                done
+ 	     fi
  	    done
  	  fi
  	  if [ "$type" = "map-t" ]; then
***************
*** 235,240 ****
--- 239,246 ----
  	proto_config_add_int "ttl"
  	proto_config_add_string "zone"
  	proto_config_add_string "encaplimit"
+ 	proto_config_add_int "snatstartps"
+ 	proto_config_add_int "snatendps"
  }
  
  [ -n "$INCLUDE_ONLY" ] || {
